File: src/lib/constants/app-identity.ts
/**
 * Identitas produk — SATU-SATUNYA tempat yang diganti saat template dipakai.
 *
 * Nilai di berkas ini bukan sekadar label. `APP_SLUG` ikut membentuk nama
 * cookie sesi dan kunci penyimpanan peramban, sehingga dua produk turunan yang
 * kebetulan dipasang pada host yang sama tidak saling menimpa sesinya. Versi
 * sebelumnya menyebar nilai-nilai ini sebagai literal di banyak berkas, dan
 * `rename-project.ts` tidak menyentuhnya — akibatnya setiap produk baru
 * membawa identitas produk lain tanpa ada yang menyadarinya.
 *
 * Padanan sisi Rust-nya ada di `src-tauri/src/desktop/app_identity.rs`, dan
 * `bun run rename` mengganti keduanya sekaligus.
 */

/**
 * Nama mesin produk: huruf kecil, angka, dan tanda hubung.
 *
 * Dipakai membentuk pengenal yang harus unik antar produk, bukan untuk
 * ditampilkan kepada pengguna.
 */
export const APP_SLUG = "app-template";

/** Nama yang dilihat pengguna. */
export const APP_DISPLAY_NAME = "App Template";

/**
 * Alamat server aplikasi bawaan, dipakai sebelum pengguna mengonfigurasinya.
 *
 * SENGAJA kosong pada template. Mengisinya dengan alamat nyata berarti setiap
 * produk turunan diam-diam menunjuk deployment milik orang lain sampai
 * seseorang menyadarinya.
 */
export const DEFAULT_SERVER_ORIGIN = "";

/** Nama cookie sesi Web. Wajib berbeda antar produk pada host yang sama. */
export const WEB_SESSION_COOKIE = `${APP_SLUG.replace(/-/g, "_")}_session`;

/** Awalan kunci `localStorage`, dengan alasan yang sama seperti cookie. */
export const STORAGE_PREFIX = APP_SLUG;

File: src-tauri/src/desktop/app_identity.rs
//! Identitas produk — SATU-SATUNYA tempat yang diganti saat template dipakai.
//!
//! Nilai di sini bukan label. [`APP_SLUG`] ikut membentuk **pemisah domain
//! kriptografi**: ia masuk ke turunan kunci vault dan ke kunci identitas
//! offline. Dua produk yang berbagi nilai ini berbagi ruang nama kriptografi
//! yang justru seharusnya memisahkan mereka.
//!
//! Versi sebelumnya menyebar nilai-nilai ini sebagai literal `"contoh-..."` di
//! beberapa berkas, dan `rename-project.ts` tidak menyentuhnya — sehingga
//! setiap produk turunan membawa pemisah domain milik produk lain.
//!
//! Padanan sisi TypeScript-nya ada di `src/lib/constants/app-identity.ts`, dan
//! `bun run rename` mengganti keduanya sekaligus.

/// Nama mesin produk: huruf kecil, angka, dan tanda hubung.
pub const APP_SLUG: &str = "app-template";

/// Nama yang dilihat pengguna.
#[allow(dead_code)]
pub const APP_DISPLAY_NAME: &str = "App Template";

/// Alamat server aplikasi bawaan.
///
/// SENGAJA kosong pada template. Mengisinya dengan alamat nyata berarti setiap
/// produk turunan diam-diam menunjuk deployment milik orang lain sampai
/// seseorang menyadarinya.
pub const DEFAULT_SERVER_ORIGIN: &str = "";

/// Pengenal klien vault Stronghold.
pub fn vault_client_id() -> Vec<u8> {
    format!("{APP_SLUG}-desktop-auth-v1").into_bytes()
}

/// Pemisah domain untuk kunci identitas offline.
///
/// Diakhiri titik dua supaya bagian yang menyusul tidak pernah bisa dibaca
/// sebagai kelanjutan slug — dua produk bernama `pos` dan `pos-lite` harus
/// menghasilkan hash yang berbeda meskipun sisa masukannya sama.
pub fn offline_identity_domain() -> String {
    format!("{APP_SLUG}-offline-identity-v2:")
}

/// Pemisah domain untuk frasa sandi vault konfigurasi database.
pub fn vault_master_domain() -> String {
    format!("{APP_SLUG}-vault-master-turso-v2:")
}
